aioshelly/rpc_device.py

```diff
@@ -88,13 +88,22 @@ async def initialize(self) -> None:
         try:
             self.shelly = await get_info(self.aiohttp_session, self.options.ip_address)
 
-            if self.options.auth or not self.requires_auth:
-                await self._wsrpc.connect(self.aiohttp_session)
-                await asyncio.gather(
-                    self.update_device_info(),
-                    self.update_config(),
-                    self.update_status(),
+            if self.requires_auth:
+                if not self.options.auth:
+                    raise AuthRequired
+
+                self._wsrpc.set_auth_data(
+                    self.shelly["auth_domain"],
+                    cast(str, self.options.username),
```

Comment:
Why does this need casting?

Reply:
I used the `self.options.auth`, this is why it confuses mypy, but there is no reason to use it here, changed it to 
https://github.com/home-assistant-libs/aioshelly/blob/00b286b4997ac176633d64b44ac2a51cf7273d3b/aioshelly/rpc_device.py#L92